packages/block-editor/src/components/use-editor-fonts-resolver/index.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * WordPress dependencies
+ */
+import { useState, useMemo, useCallback } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { loadFontFaceInBrowser } from './utils';
+import { store as blockEditorStore } from '../../store';
+import { globalStylesLinksDataKey } from '../../store/private-keys';
+
+function resolveThemeFontFaceSrc( src, _links ) {
+	const firstSrc = Array.isArray( src ) ? src[ 0 ] : src;
+	return _links.find( ( link ) => link.name === firstSrc )?.href || firstSrc;
+}
+
+function useEditorFontsResolver() {
+	const [ loadedFontUrls, setLoadedFontUrls ] = useState( new Set() );
+
+	const { _links = [], fontFamilies = [] } = useSelect( ( select ) => {
+		const { getSettings } = select( blockEditorStore );
+		const _settings = getSettings();
+		return {
+			_links: _settings[ globalStylesLinksDataKey ]?.[ 'wp:theme-file' ],
+			fontFamilies:
+				_settings?.__experimentalFeatures?.typography?.fontFamilies,
+		};
+	}, [] );
+
+	// eslint-disable-next-line no-console
+	console.log( 'fontFamilies', fontFamilies, 'links', _links );
+
+	const fontFaces = useMemo( () => {
+		return Object.values( fontFamilies )
+			.flat()
+			.map( ( family ) => family.fontFace )
+			.filter( Boolean )
+			.flat();
+	}, [ fontFamilies ] );
+
+	const loadFontFaceAsset = useCallback(
+		async ( fontFace, ownerDocument ) => {
+			if ( ! fontFace.src ) {
+				return;
+			}
+
+			const src = resolveThemeFontFaceSrc( fontFace.src, _links );
```

Comment:
```suggestion
			const src = getThemeFontFaceSrc( fontFace.src, _links );
```

Nit: for code clarity purposes we should probably avoid using "resolve" inside an `async` function. I thought this was async.

Reply:
This part of the code was removed. Marking this comment as resolved.